svm/src/account_saver.rs

```diff
@@ -58,100 +51,87 @@ pub fn collect_accounts_to_store<'a>(
             continue;
         };
 
-        let loaded_transaction = &mut executed_tx.loaded_transaction;
-        let execution_status = &executed_tx.execution_details.status;
-
-        // Accounts that are invoked and also not passed as an instruction
-        // account to a program don't need to be stored because it's assumed
-        // to be impossible for a committable transaction to modify an
-        // invoked account if said account isn't passed to some program.
-        //
-        // Note that this assumption might not hold in the future after
```

Comment:
This note goes away because the new implementation doesn't check for `is_storable_account` for failed transactions. This is because it's assumed that fee payer and nonce accounts are always storable because if they were invoked, the transaction would be dropped.